store/index.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { createContext } from 'react'
+import { useStaticRendering, useLocalStore, observer } from 'mobx-react'
+
+const isServer = typeof window === 'undefined'
+useStaticRendering(isServer)
+
+const GlobalContext = createContext({})
+
+export const initializeData = (initialData = {}) => {
+  return {
+    plugins: {
+      discovery: {
+        enabled: true,
+      },
+      recommender: {
+        enabled: false,
+      },
+      analytics: {
+        enabled: true,
+      },
+      reader: {
+        enabled: false,
+      },
+    },
+    ...initialData,
+  }
+}
+
+export const withGlobalStore = Component => {
+  return props => (
+    <GlobalContext.Consumer>
+      {value => {
+        const ObservableComponent = observer(Component)
+        return <ObservableComponent {...props} store={value} />
```

Comment:
I would change the order. If somebody redefines `store` prop, it's by purpose, IMHO.